BSP/Simulator/Src/BSP_ADC.c

```diff
@@ -0,0 +1,100 @@
+/*
+C file that reads the raw ADC data from the CSV file, 
+converts it back to a distance pressed (percentage for now), and stores both.
+We should be able to request both the raw ADC data,
+the distance (percentage for now) and millivolts from this library.
+ */
+
+//NOTE: ADC precision is 12 bits (4096), range 3.3V
+//I'm assuming 3.3V corresponds to a 100% pressed pedal
+
+#include "BSP_ADC.h"
+
+#define FILE_NAME DATA_PATH(PEDALS_CSV)
+
+/**
+ * @brief   Confirms that the CSV file
+ *          has been created and prints and 
+ *          error message if not
+ * @param   None
+ * @return  None
+ */ 
+void BSP_ADC_Init(void) {
+    // Checking if file exists
+   if(access(FILE_NAME, F_OK) != 0){
+        // File doesn't exist if true
+        perror(PEDALS_CSV);
+        exit(EXIT_FAILURE);
+    }
+    // There really isn't much we have to do in the simulator
+}
+
+/**
+ * @brief   Helper function which returns a 
+ *          milivoltage based on the raw ADC value
+ *          compared to the 3.3 volt range of the 
+ *          pedals' potentiometers
+ * @param   raw ADC value
+ * @return  milivolts 
+ */ 
+int16_t convert_ADC_to_Voltage(int16_t ADCvalue){
+    return (((float)ADCvalue / 4096) * 3300);
```

Comment:
Casting to a `float` is taking an unnecessary performance hit, since the function ultimately returns an int16_t. What would be better is something along the lines of what's shown below.
```suggestion
    return (uint16_t) (((uint32_t)ADCvalue * 3300) / 4096);
```
The cast to `uint32_t` ensures that the multiplication does not overflow (and since the registers are 32-bit, shouldn't be different in terms of performance), and moving the division to the last step avoid early truncation, without having to involve the FPU. The final cast to uint16_t is optional, since it should be implicitly done anyways, but it's never a bad idea to be explicit about things.

Also, I don't like magic numbers.

Reply:
Yeah we want to avoid using floating point numbers as much as possible. Since this is in mV, we don't need to have a fraction portion of our number for the precision we want. 

As for the magic numbers, I would go ahead and put some #defines in the header file to explicitly say what these numbers are